collector/mongos/collections_status.go

```diff
@@ -100,9 +103,9 @@ var (
 )
 
 // GetCollectionStatList returns stats for a given database
-func GetCollectionStatList(session *mgo.Session) *CollectionStatList {
+func GetCollectionStatList(ctx mongo.SessionContext, client *mongo.Client) *CollectionStatList {
```

Comment:
All mongod functions were changed to accept `*mongo.Client` instead of `*mgo.Session`. All mongos function were changed to also accept `mongo.SessionContext`. Is there a reason for that? And if yes, is there a reason why `context.TODO()` is used below instead of `ctx`?

Reply:
> All mongos function were changed to also accept mongo.SessionContext. Is there a reason for that?

Yes, here is the reason:

Old implementation:

https://github.com/percona/mongodb_exporter/blob/0dcd9d7b7e0c8d58ed44dbca726e1b58c0da2106/collector/mongodb_collector.go#L238-L240

New implementation:

https://github.com/percona/mongodb_exporter/blob/156801d2b09738a522e620a94cd1ea5af63545cc/collector/mongodb_collector.go#L240-L243

Reply:
> That hardly explains it. https://jira.mongodb.org/browse/SERVER-27864 was fixed two years ago in a version that is no longer supported.
> I think we should update all functions to accept only mongo.SessionContext

Then it's not necessary to use `mongo.SessionContext` at all. Especially because [it's not concurrent safe.](https://godoc.org/go.mongodb.org/mongo-driver/mongo#SessionContext).

> and also get a client with ctx.Client().
There is no `ctx.Client()` in driver version `1.0.2` only in `master`.

I propose to remove this context at this time and add context in the next task. Also, I don't like that exporter use global variables through all codebase, it seems we should refactor it.

Reply:
> I propose to remove this context at this time

Ok, let's accept only Client everywhere.

Agree with other points.